python/lsst/sims/photUtils/SedList.py

```diff
@@ -0,0 +1,391 @@
+
+        @param [in] fileDir is the base directory where the Sed files are stored
+        (defaults to LSST sims_sed_library package).
+
+        @param [in] specMap is a specMap (defined in sims_utils/../fileMaps.py)
+        that maps the names in sedNameList to paths of the files relative to
+        fileDir (defaults to defaultSpecMap defined in sims_utils)
+
+        @param [in] wavelenMatch is an optional numpy array representing
+        the wavelength grid to which all Seds will be re-mapped.
+
+        @param [in] redshiftList is an optional list of redshifts for the Sed
+
+        @param [in] internalAvList is an optional list of A(V) due to internal
+        dust (for spectra of galaxies).
+
+        @param [in] galacticAvList is an optional list of A(V) due to
+        Milky Way Dust.
+
+        @param [in] cosmologicalDimming is a boolean indicating whether cosmological
+        dimming (the extray (1+z)^-1 factor in flux) should be applied to spectra
+        when they are redshifted (defaults to True)
+
+        Note: once wavelenMatch and cosmologicalDimming have been set in
+        the constructor, they cannot be un-set.
+
+        Similarly: if you construct a SedList without a galacticAvList,
+        internalAvList, or redshiftList, you cannot later add spectra with
+        whichever of those features were left out.
+        """
+
+
+        self._initialized = False
+        self._spec_map = specMap
+        self._wavelen_match = copy.deepcopy(wavelenMatch)
+        self._file_dir = fileDir
+        self._cosmological_dimming = cosmologicalDimming
+
+        #self._unique_sed_dict will store all of the unique SED files that have been
+        #loaded.  If an object requires an SED that has already been loaded,
+        #it will just copy it from the dict.
+        self._unique_sed_dict = {}
+        self._unique_sed_dict['None'] = Sed()
+
+        # initialize the delat funciton bandpass for calibrating
+        # magNorm
+        self._imsimband = Bandpass()
+        self._imsimband.imsimBandpass()
+
+        self._sed_list = []
+        self._redshift_list = None
+        self._galactic_av_list = None
+        self._internal_av_list = None
+
+        self._a_int = None
+        self._b_int = None
+        self._av_int_wavelen = None
+
+        self._a_gal = None
+        self._b_gal = None
+        self._av_gal_wavelen = None
+
+        self.loadSedsFromList(sedNameList, magNormList,
+                               internalAvList = internalAvList,
+                               galacticAvList = galacticAvList,
+                               redshiftList = redshiftList)
+
+
+    def __len__(self):
+        return len(self._sed_list)
+
+    def __getitem__(self, index):
+        return self._sed_list[index]
+
+    def __iter__(self):
+        for val in self._sed_list:
+            yield val
+
+
+    # Handy routines for handling Sed/Bandpass routines with sets of dictionaries.
```

Comment:
This loads the seds into sedlists during instantiation. Is there an expectation that this will be used to append more seds at a later stage?